### PR TITLE
Fix crash if bones punched by non-player

### PR DESCRIPTION
**Pull request by berengma**
- Added check for player:is_player()
- if bones are fresh and the bones are punched nothing happens
- if bones are old, the inventory is dropped as entities and the bones node is removed

### CODE REVIEW

**Review by SmallJoker — COMMENTED** [review]
Works. Tested with the code in https://github.com/minetest/minetest_game/issues/3145#issuecomment-2254163679.

File: mods/bones/init.lua [hunk 1]
```diff
@@ -16,6 +16,27 @@ local function is_owner(pos, name)
 	return false
 end
 
+local drop = function(pos, itemstack)
```
Comment (appgurueu) [inline mods/bones/init.lua:19]:
```suggestion
local function drop(pos, itemstack)
```

File: mods/bones/init.lua [hunk 2]
```diff
@@ -16,6 +16,27 @@ local function is_owner(pos, name)
...
+	if obj then
+		obj:set_velocity({
+			x = math.random(-10, 10) / 9,
+			y = 5,
+			z = math.random(-10, 10) / 9,
+		})
+	end
+end
+
+local dropContents = function(pos)
```
Comment (appgurueu) [inline mods/bones/init.lua:30]:
```suggestion
local function drop_contents(pos)
```

File: mods/bones/init.lua [hunk 3]
```diff
@@ -87,6 +108,10 @@ local bones_def = {
 			return
 		end
 
+		if not player:is_player() then
+			dropContents(pos)
```
Comment (appgurueu) [inline mods/bones/init.lua:112]: Shouldn't this be followed by a `return`?

File: mods/bones/init.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (berengma) [reply mods/bones/init.lua]: Good to know that there are code style guidelines and where. I changed it.

File: mods/bones/init.lua (hunk 1 above)
Comment (berengma) [reply mods/bones/init.lua:19]: I cannot find anything about this in the guidelines. Furthermore this function had been there before. :)

File: mods/bones/init.lua (hunk 2 above)
Comment (berengma) [reply mods/bones/init.lua]: see above

File: mods/bones/init.lua (hunk 3 above)
Comment (berengma) [reply mods/bones/init.lua]: return is added now.

**Comment by berengma** [comment]
Updated PR according to code style guidelines

**Review by appgurueu — APPROVED** [review]
Tested using SmallJoker's code by setting the bone share time to a few seconds, then `/killme`, then `/p` (nothing happens, bones are still fresh), then after a few seconds `/p` again (items are dropped).

File: mods/bones/init.lua (hunk 1 above)
Comment (appgurueu) [reply mods/bones/init.lua:19]: Well the function right above it uses the `local function` syntactic sugar. (I also prefer this style and it costs nothing to apply it consistently as we go.)

**Comment by berengma** [comment]
Anyone who could do a second review ?

**Review by berengma — COMMENTED** [review]
from my point everything is resolved

**Review by SmallJoker — APPROVED** [review]

File: mods/bones/init.lua [hunk 4]
```diff
@@ -16,6 +16,27 @@ local function is_owner(pos, name)
...
+	if obj then
+		obj:set_velocity({
+			x = math.random(-10, 10) / 9,
+			y = 5,
+			z = math.random(-10, 10) / 9,
+		})
+	end
+end
+
+local drop_contents = function(pos)
```
Comment (sfan5) [inline mods/bones/init.lua:30]:
```suggestion
local function drop_contents(pos)
```

**Pull request merged**